### PR TITLE
Project: todolist structure

### PR DESCRIPTION
**Pull request by wwwdrt**
# 👋 Hey CRs

> I'm looking forward to your feedback 🚀

Please feel free to reach out to me on Slack anytime. You can send me a direct message or give me a call. If you choose to call, please introduce yourself so I know who I'm speaking with. I'm here to help, so don't hesitate to get in touch 😁

<div align="center">
  
</div>

# ⚒️ Development

> Clear explanation of the approach I took in the project 👇

## 🗒️Getting Started

I began by creating a placeholder for the todolist in the `index.html` file. Following that, I developed a class component called `Data.js`. Inside this component, I defined a private field. In the constructor, I initialized this private field as an empty array of objects. Each object in the array consists of three properties: a string, a boolean value, and an index.

Next, I implemented a method to render each task in the todolist. To achieve this, I utilized a forEach loop to iterate over each element in the array of objects. During the iteration, I employed conditional statements to determine whether a task is completed or in progress. Based on the task's status, I updated the textContent accordingly. Finally, I appended the rendered task to the HTML document.

## 🗒️Webpack

> The approach I took at configuring webpack 👇

In addition to the basic setup, I made some additional configurations to enhance the development experience. Firstly, I included the inline-source-map option, which helps track down errors by providing detailed information about their origins in the source code.

Moreover, I implemented the clean rule for the output files. This rule ensures that only the necessary files are included in the distribution code, removing any unused or unnecessary files. This helps optimize the final output and keep it clutter-free.

## 🗒️Styles

> How I handled style.css import 👇

My `Data.js` component is responsible for rendering the tasks and it is the component that interacts directly with the DOM elements, that's why I decided to `import styles` in the component.

So, when I create an instance of the Data class in my `index.js` file (`new Data()`), the styles get applied to the tasks rendered by the Data component.

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -3,15 +3,18 @@
...
-
-<h1>TEST</h1>
-
-<script src="/src/index.js"></script>
+<div id="wrapper">
+    <div id="root">
+        <ul id="todolist">
+        </ul>
+    </div>
+</div>
```
Comment (thecodechaser) [inline dist/index.html:17]: - [ ] Your to-do list structure is missing most of the components, for example, the heading, form field, checkboxes, and clear all completed button. Please make sure to add them and your design should match the given design.

### Your design

image

### Given design

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @wwwdrt👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: dist/index.html (hunk 1 above)
Comment (wwwdrt) [reply dist/index.html:17]: Hey @thecodechaser, changes made ✅ 

image

**Review by OyePriscilla — APPROVED** [review]
Hi @wwwdrt 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Design according to the wireframe ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**